Firmware/ultralcd.cpp

```diff
@@ -4562,25 +4544,26 @@ static void lcd_calibration_menu()
 #endif //TMC2130
     MENU_ITEM_FUNCTION_P(_i("Selftest"), lcd_selftest_v);////MSG_SELFTEST c=18
     // MK2
-    MENU_ITEM_FUNCTION_P(_i("Calibrate XYZ"), lcd_mesh_calibration);////MSG_CALIBRATE_BED c=18
+    MENU_ITEM_GCODE_P(_i("Calibrate XYZ"), PSTR("M45"));////MSG_CALIBRATE_BED c=18
     // "Calibrate Z" with storing the reference values to EEPROM.
-    MENU_ITEM_SUBMENU_P(_T(MSG_HOMEYZ), lcd_mesh_calibration_z);
+    MENU_ITEM_GCODE_P(_T(MSG_HOMEYZ), PSTR("M45 Z"));
 
-    MENU_ITEM_SUBMENU_P(_T(MSG_MESH_BED_LEVELING), lcd_mesh_bedleveling); ////MSG_MESH_BED_LEVELING c=18
+    MENU_ITEM_GCODE_P(_T(MSG_MESH_BED_LEVELING), PSTR("G80")); ////MSG_MESH_BED_LEVELING c=18
 
     MENU_ITEM_SUBMENU_P(_i("Bed level correct"), lcd_adjust_bed);////MSG_BED_CORRECTION_MENU c=18
     MENU_ITEM_SUBMENU_P(_i("PID calibration"), pid_extruder);////MSG_PID_EXTRUDER c=17
 #ifndef TMC2130
     MENU_ITEM_SUBMENU_P(_i("Show end stops"), menu_show_end_stops);////MSG_SHOW_END_STOPS c=18
 #endif
-    MENU_ITEM_GCODE_P(_i("Reset XYZ calibr."), PSTR("M44"));////MSG_CALIBRATE_BED_RESET c=18
+    MENU_ITEM_SUBMENU_P(_T(MSG_CALIBRATE_BED_RESET), lcd_calibration_bed_reset);
 #ifdef PINDA_THERMISTOR
     if(has_temperature_compensation())
+        //Keep as function menu to show status screen and process of the calibration
         MENU_ITEM_FUNCTION_P(_T(MSG_PINDA_CALIBRATION), lcd_calibrate_pinda);
 #endif
 
 #ifdef THERMAL_MODEL
-    MENU_ITEM_SUBMENU_P(_n("Thermal Model cal."), lcd_thermal_model_cal);
+    MENU_ITEM_FUNCTION_P(_n("Thermal Model cal."), lcd_thermal_model_cal);
```

Comment:
This is a good change. The behaviour was wrong before, but it was saved by the lcd_return_to_status() in the function.